### PR TITLE
chore(ts): Convert sidebar

### PR DESCRIPTION
**Pull request by evanpurkhiser**

### CODE REVIEW

File: src/sentry/static/sentry/app/components/sidebar/index.tsx [hunk 1]
```diff
@@ -624,29 +645,31 @@ class Sidebar extends React.Component {
...
-    return {
-      collapsed: PreferencesStore.getInitialState().collapsed,
-    };
-  },
-
-  onPreferenceChange(store) {
-    if (store.collapsed === this.state.collapsed) {
-      return;
-    }
+const SidebarContainer = ReactRouter.withRouter(
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/components/sidebar/index.tsx:648]: Hmm, my conversion here is from a somewhat old stash, so maybe we removed react router recently.

**Review by matejminar — APPROVED** [review]

File: src/sentry/static/sentry/app/components/sidebar/index.tsx (hunk 1 above)
Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/components/sidebar/index.tsx:648]: Ahh I see billy did this here 84df9403ff33106cc8a43d30966efa5209eb7a20

**Comment by matejminar** [comment]
Apart from that TS error LGTM

**Comment by EvanPurkhiser** [comment]
Yeah fixing that up

**Pull request merged**